src/lib/radio/radio/radio-foundation.ts

```diff
@@ -0,0 +1,233 @@
+import { RADIO_CONSTANTS, RadioLabelPosition } from './radio-constants';
+
+export interface IRadioFoundation extends ICustomElementFoundation {
+  checked: boolean;
+  defaultChecked: boolean;
+  value: string;
+  disabled: boolean;
+  required: boolean;
+  readonly: boolean;
+  dense: boolean;
+  labelPosition: RadioLabelPosition;
+  tryCheck(): boolean;
+}
+
+export class RadioFoundation implements IRadioFoundation {
+  // State
+  private _checked = false;
+  private _defaultChecked = false;
+  private _value = 'on';
+  private _disabled = false;
+  private _required = false;
+  private _readonly = false;
+  private _dense = false;
+  private _labelPosition: RadioLabelPosition = 'end';
+
+  // Listeners
+  private readonly _focusListener: EventListener;
+  private readonly _blurListener: EventListener;
+  private readonly _clickListener: EventListener;
+  private readonly _keydownListener: EventListener;
+  private readonly _keyupListener: EventListener;
+
+  constructor(private _adapter: IRadioAdapter) {
+    this._focusListener = () => this._handleFocus();
+    this._blurListener = () => this._handleBlur();
+    this._clickListener = (evt: MouseEvent) => this._handleClick(evt);
+    this._keydownListener = (evt: KeyboardEvent) => this._handleKeydown(evt);
+    this._keyupListener = (evt: KeyboardEvent) => this._handleKeyup(evt);
+  }
+
+  public initialize(): void {
+    this._adapter.addHostListener('focus', this._focusListener);
+    this._adapter.addHostListener('blur', this._blurListener);
+    this._adapter.addHostListener('click', this._clickListener);
+    this._adapter.addHostListener('keydown', this._keydownListener);
+    this._adapter.addHostListener('keyup', this._keyupListener);
+  }
+
+  // Public methods
+
+  public tryCheck(): boolean {
+    this._checked = true;
+
+    if (!this._dispatchEvents()) {
+      this._checked = false;
+      return false;
+    }
+    
+    this._adapter.setChecked(this._checked, this._value);
+    this._adapter.toggleHostAttribute(RADIO_CONSTANTS.attributes.CHECKED, this._checked);
+    return true;
+  }
+
+  // Event handlers
+
+  private _handleFocus(): void {
+    if (!this._checked) {
+      this._adapter.setUncheckedRadioGroupFocus('focus');
+    }
+  }
+
+  private _handleBlur(): void {
+    if (!this._checked) {
+      this._adapter.setUncheckedRadioGroupFocus('blur');
+    }
+  }
+
+  private _handleClick(evt: MouseEvent): void {
+    this._activate(evt);
+  }
+
+  private _handleKeydown(evt: KeyboardEvent): void {
+    switch (evt.key) {
+      case 'ArrowRight':
+      case 'ArrowDown':
+        evt.preventDefault();
+        this._adapter.focusNext();
+        break;
+      case 'ArrowLeft':
+      case 'ArrowUp':
+        evt.preventDefault();
+        this._adapter.focusPrevious();
+        break;
+    }
+  }
+
+  private _handleKeyup(evt: KeyboardEvent): void {
+    if (evt.key === ' ') {
+      evt.preventDefault();
+      this._activate(evt);
+    }
```

Comment:
When testing the component I noticed that when no radios were checked, tabbing into the first one, and pressing space did not seem to check the radio.